docs/kb/integrations/defect-dojo-integration.md

```diff
@@ -0,0 +1,114 @@
+
+So a Python function using this endpoint can look like this:
+
+```
+def uploadToDefectDojo(is_new_import, token, url, product_name, engagement_name, filename):
+    multipart_form_data = {
+        'file': (filename, open(filename, 'rb')),
+        'scan_type': (None, 'Semgrep JSON Report'),
+        'product_name': (None, product_name),
+        'engagement_name': (None, engagement_name),
+    }
+
+    endpoint = '/api/v2/import-scan/' if is_new_import else '/api/v2/reimport-scan/'
+    r = requests.post(
+        url + endpoint,
+        files=multipart_form_data,
+        headers={
+            'Authorization': 'Token ' + token,
+        }
+    )
+    if r.status_code != 200:
+        sys.exit(f'Post failed: {r.text}')
+    print(r.text)
+```
+
+The full version of this Python script can be found in the [Semgrep Utilities repository](https://github.com/r2c-CSE/semgrep-utilities/blob/main/integrations/defectdojo/import_semgrep_to_defect_dojo.py).
+It is ready to execute.
+
+### Running the script
+
+To run the script:
+```
+python3 integrations/defectdojo/import_semgrep_to_defect_dojo.py --host DOJO_URL --product PRODUCT_NAME --engagement ENGAGEMENT_NAME --report REPORT_FILE 
+```
+Where:
+* `DOJO_URL` is the URL where DefectDojo is.
+* `PRODUCT_NAME` is the DefectDojo product name.
+* `ENGAGEMENT_NAME` is the DefectDojo engagement name for that product.
+* `REPORT_FILE` is the Semgrep report path
+
+## Integrating Semgrep and DefectDojo in a CI pipeline
+
+To prevent tampering at any point of delivery, it is crucial to integrate security scans and dump results to DefectDojo in the **same pipeline or CI job**.
+The following is an example of a GitLab job importing Semgrep findings to DefectDojo:
+
+```
+import-semgrep-to-defectdojo:
+  stage: import
+  image: python:3.9-bullseye
+  script:
+    - echo "Importing Semgrep scan to DefectDojo"
+    - pip3 install requests
+    - curl -O https://raw.githubusercontent.com/r2c-CSE/semgrep-utilities/main/integrations/defectdojo/import_semgrep_to_defect_dojo.py
+    - python3 import_semgrep_to_defect_dojo.py --host $DEFECTDOJO_URL --product $PRODUCT --engagement semgrep --report report.json || true
+  rules:
+    # Scan changed files in MRs, (diff-aware scanning):
+    - if: $CI_MERGE_REQUEST_IID
+    # Scan mainline (default) branches and report all findings.
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+  tags:
+    - defectdojo
+  variables:
+    DEFECT_DOJO_API_TOKEN: $DEFECT_DOJO_API_TOKEN
+```
+
+### Adding checksum validation
+During the CI job and to verify that the job downloads the correct Python script securely, check the integrity with a checksum such as SHA 256 code. It prevents tampered scripts from executing.
+
+```
+    - echo $IMPORT_SEMGREP_TO_DEFECTDOJO_SHA_CHECKSUM > sha-import-dd.tmp
+    - shasum -a 256 -U -c sha-import-dd.tmp
+```
+Where:
+* `IMPORT_SEMGREP_TO_DEFECTDOJO_SHA_CHECKSUM` is the result of executing `shasum -a 256 -U import_semgrep_to_defect_dojo.py`
```

Comment:
While this is a good practice, I'm not sure this section is necessary or super relevant to the article topic. What do we think about taking it out entirely? Could be a little confusing the reader maybe? I would be curious on other's thoughts as well (cc @armchairlinguist).

Reply:
If we think folks should do this, I would suggest just including it in the example and explaining briefly why it's included, something like:

"This pipeline includes checksum validation for the import script, to ensure that the script has not changed unexpectedly."

I'm a little bit unclear on what the actual process is, though. Where is this $IMPORT... coming from? So we might need more than the one sentence of explanation.